### DISCUSSION TITLE
clangd doesn't know about GCC "-fanalyzer"

### DISCUSSION THREAD

**Discussion by superjamie** [clangd/clangd#2121] [Ideas]
If the compilation database includes GCC's `-fanalyzer`, an error is inserted in every file because clangd expects clang's `-Xanalyzer`.

Is it possbible to maybe alias `-fanalyzer` to `-Xanalyzer`, or mask this error?

I understand it's "clang daemon" and is intended for "clang compiler" but it's also common to use GCC, and some environments are not easy/possible to use clang, such as DJGPP or other toolchains which depend on GCC.

**Comment by HighCommander4** [comment]
> Is it possbible to maybe alias `-fanalyzer` to `-Xanalyzer`

You can make these tweaks (and others) in the [config file](https://clangd.llvm.org/config.html#compileflags).

**Comment by superjamie** [comment]
Oh excellent.

Setting the compiler to gcc didn't work, eg:

```yaml
CompileFlags:
  Compiler: gcc
```

However just ignoring the specific flag worked:

```yaml
CompileFlags:
  Remove: -fanalyzer
```

That solves this for me. Many thanks!
